### DISCUSSION TITLE
Welcome to codebookr Discussions!

### DISCUSSION THREAD

**Discussion by mbcann01** [brad-cannell/codebook#26] [Q&A]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

**Comment by zappycoder** [comment]
Hello, 

I am using codebook in an attempt to generate a codebook for a variety of variable types. One type specifically is a Likert scale variable 1-9, but only has anchor terms for the 1, 3, 5, 7, and 9 ratings. In attempting to make a codebook including this variable, I received the following warning message:

"Warning message:
attributes are not identical across measure variables;
they will be dropped"

I believe this has something to do with the anchor terms/labels being dropped from my Likert scale variable. Is there a work around for this where I may still retain the variable labels? 

For some more context, it is a thirst rating variable I am using, and I added the following code but to no avail:

df %>% cb_add_col_attributes(
  morningthirstmean,
  col_type=integer,
  description = "Thirst Scale",
  source = "Thirst Questionnaire",
  value_labels = c(
    "Not Thirsty at ALL" = 1,
    " "=2,
    "A Little Thirsty" = 3,
    " "=4,
    "Moderately Thirsty" = 5,
    " "=6,
    "Very Thirsty" = 7,
    " "=8,
    "Very Very Thirsty" = 9
  )) 


Thank you in advance!
Mitch

**Reply by mbcann01** [reply]
Hi @zappycoder ,

Thanks for bringing up this issue. When playing around with the built-in study data, I got an error from the following code.

```
# Load packages
library(codebookr)
library(dplyr, warn.conflicts = FALSE)
library(haven)

# Load data
data(study)

# Add value labels to the Likert column
study_2 <- study %>% 
  cb_add_col_attributes(
    likert,
    value_labels = c(
      "Very dissatisfied" = 1, 
      "" = 2,
      "Neither satisfied nor dissatisfied" = 3,
      "" = 4,
      "Very satisfied" = 5
    )
  )
```
Error: `Error: attempt to use zero-length variable name`. 

This isn't a codebook error. This is a base R error. In other words, you get the same error from this code.

```
c("test" = 1, "" = 2)
```

A simple workaround is to just add a single space in between your quotation marks like this.

```
study_2 <- study %>% 
  cb_add_col_attributes(
    likert,
    value_labels = c(
      "Very dissatisfied" = 1, 
      " " = 2,
      "Neither satisfied nor dissatisfied" = 3,
      " " = 4,
      "Very satisfied" = 5
    )
  )
```

And here is the result I got:

<img width="648" alt="Screenshot 2023-03-09 at 10 52 21 AM" src="https://user-images.githubusercontent.com/12264987/224097716-9278ea2c-07dd-4a37-a717-a23b99e6496a.png">

 Does this solve your problem?
